components/dropdown/Dropdown.tsx

```diff
@@ -47,16 +51,15 @@ const Dropdown: React.FC<DropdownProps> = ({
     }
   };
 
-  const handleClick = (e: ChangeEvent<HTMLInputElement>) => {
-    e.preventDefault();
+  const chooseSuggestion = (suggestion: string) => {
     setSuggestions([]);
-    setValue(e.target.innerText);
```

Comment:
Instead of reading the innertext of the button, I just pass in the string (which city the user selected)